infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt

```diff
@@ -1267,12 +1267,15 @@ private fun getSegmentIndexForSwitchJointMatch(
     val alignment = trackAlignments.getValue(match.locationTrackId)
     val segmentIndex = alignment.getSegmentIndexAtM(match.m)
     val segment = alignment.segments[segmentIndex]
+    // The m-value for a suggested switch is very often that of a segment start or end. Since switch suggestions are
+    // created for all split targets on a track before they're used for relinking, we can't record segment indices in
+    // switch suggestions. Allow for a little bit of wiggle room in case segment splits alter downstream m-values.
```

Comment:
Käytetäänkös tätä funkkaria myös muussakin linkityksessä kuin pelkässä splittauksessa? Tämähän taitaa olla aika geneerinen linkitysfunktio. Mietin että a) tämä kommentti saattaa olla niihin käyttökeisseihin vähän out of place ja b) niihin muihin tapauksiin saatettaisiin haluta eri deltat. Jos tätä käytetään muualla, niin olisiko järkevää esim. tehdä tästä kokonaan oma versio splittaukselle (jos sinne halutaan tuo kommentti mukaan,) tai sitten parametrisoida tuo delta?

Reply:
Ihan hyvä huomio. Toisaalta tuon kommentin voisi generalisoida vain puhumaan tuosta massa-uudelleenlinkityksestä koska siitähän tuossa on oikeastaan kyse enemmänkin kuin splittauksesta (uudelleenlinkitys tapahtuu ensiksi ja se on se mikä segment indeksit rikkoo). Tuo kommentti voisi olla ehkä paremmalla paikalla vaihde-ehdotuksessa, jotta siitä näkisi että se on kuvattu m-arvolla indeksin sijaan tästä syystä. 

Vastaavasti täällä on sitten oikeastaan enää kyse siitä että koska m-arvo on liukuluku, sitä ei voi == vertailla mihinkään vaan täytyy snapata match-tyypin mukaan.